[Partial] Adds API token authentication

jaenia opened the pull request:
[As a Programmer I write a RESTful endpoint to server the React Frontend](https://trello.com/c/OpftBl6x/26-as-a-programmer-i-write-a-restful-endpoint-to-server-the-react-frontend)

### Description
This PRs authenticates users with a token.

### How to test
- [ ] Go to `http://localhost:8000/api/users/token`
- [ ] Check that if you provide valid credentials, you'll receive an authentication token
- [ ] Check that if you provide invalid credentials, you'll receive a response status of 400 and the message "Unable to authenticate with provided credentials"

jaenia committed:
Adds Django Rest Framework

jaenia committed:
Adds token authentication

jaenia committed:
Implements serializers.Serializer asbtract methods

jaenia committed:
Renames test class for create token view

jaenia committed:
Moves users api from users app to api folder

jaenia committed:
Renames CreateTokenView to CreateTokenEndpoint

jaenia committed:
Renames users api url

Replaces user by users

jaenia committed:
Fixes prospector errors

jaenia committed:
Removes unused battle api references

jaenia committed:
Renames CreateTokenEndpoint to LoginEndpoint

jaenia committed:
Return a created token related to the user

filipeximenes commented on backend/api/users/views.py:
```diff
@@ -0,0 +1,10 @@
+from api.users.serializers import AuthTokenSerializer
+
+
+class LoginEndpoint(ObtainAuthToken):
```
It looks like the `ObtainAuthToken` already handles Token creation, you only need to ensure there's a `user` in the `validated_data` of the serializer: https://github.com/encode/django-rest-framework/blob/master/rest_framework/authtoken/views.py#L54

jaenia replied:
Awesome!

filipeximenes commented on backend/api/users/serializers.py:
```diff
@@ -0,0 +1,26 @@
+        attrs["user"] = user
+        return attrs
+
+    def create(self, validated_data):
+        return Token.objects.get_or_create(**validated_data)
+
+    def update(self, instance, validated_data):
+        return instance
```
You can probably delete these

jaenia replied:
Good to know :)

jaenia committed:
Removes unnecessary create and update methods from AuthTokenSerializer

jaenia committed:
Ignores lint on AuthTokenSerializer

By not overriding create and update methods, pylint - abstract-method is raised

jaenia committed:
Fixes black errors
